### PR TITLE
Add autoFocus prop

### LINKED ISSUE #276: How can I set focus on the input element from the outside?
I'd like to autofocus the input element and show the calendar once the DatePicker is rendered, in order to allow the user to select a date w/o forcing him to click on the input field first. Is there a way?

### PR DESCRIPTION
**Pull request by MarkReeder**
Fixes #276

### CODE REVIEW

File: src/datepicker.jsx [hunk 1]
```diff
@@ -92,6 +93,16 @@ var DatePicker = React.createClass({
     }
   },
 
+  componentDidMount () {
+    if (this.props.autoFocus) {
```
Comment (rafeememon) [inline src/datepicker.jsx]: Did you try simply passing the `autoFocus` prop to the `input` element?  It would be nice to delegate this to native HTML if possible.

File: test/datepicker_test.js [hunk 2]
```diff
@@ -298,4 +298,13 @@ describe('DatePicker', () => {
     var result = data.callback.args[0][0]
     expect(result.format(data.testFormat)).to.equal(data.copyM.format(data.testFormat))
   })
+  it('should autofocus the input given the autoFocus prop', (done) => {
+    var div = document.createElement('div')
+    document.body.appendChild(div)
+    ReactDOM.render(<DatePicker autoFocus />, div)
+    defer(() => {
```
Comment (rafeememon) [inline test/datepicker_test.js]: Out of curiosity, can this `defer` go away?  Not sure if focusing happens sync or async

Comment (MarkReeder) [inline test/datepicker_test.js]: Good call- the test failed previously, but using the autoFocus prop, it's not necessary. I'll remove.

File: src/datepicker.jsx [hunk 3]
```diff
@@ -92,6 +93,10 @@ var DatePicker = React.createClass({
     }
   },
 
+  setFocus () {
```
Comment (rafeememon) [inline src/datepicker.jsx]: (sorry for the back-and-forth, hard to give a complete CR on the train =P)  It looks like this part of the change can probably go away

Comment (rafeememon) [inline src/datepicker.jsx]: Agreed that it might be useful, but that should be addressed in another PR to separate concerns =)

**Comment by rafeememon** [comment]
lgtm, thanks!
